### PR TITLE
crash report on mac little update.

### PR DESCRIPTION
**Pull request by devnexen**
displaying vm info as Linux and FreeBSD.
checking libproc as it is present only from 10.5 version.

### CODE REVIEW

File: vm_dump.c [hunk 1]
```diff
@@ -984,6 +991,41 @@ rb_vm_bugreport(const void *ctx)
...
+		if (vm_region_recurse_64(mach_task_self(), &addr, &size, &depth,
+			(vm_region_recurse_info_t)&map, &count) != KERN_SUCCESS) {
+			break;
+		}
+
+		if (map.is_submap) {
+			// We only look at main addresses
+			depth++;
+		} else {
+			fprintf(stderr, "%lu-%lu %s%s%s", addr, (addr+size),
```
Comment (nobu) [inline vm_dump.c]: `%lx` would be better, I think.

File: vm_dump.c [hunk 2]
```diff
@@ -984,6 +991,41 @@ rb_vm_bugreport(const void *ctx)
 	    fprintf(stderr, "\n");
 	}
 #endif /* __FreeBSD__ */
+#ifdef __APPLE__
+	vm_address_t addr = 0;
+	vm_size_t size = 0;
+	struct vm_region_submap_info_64 map;
```
Comment (nobu) [inline vm_dump.c]: Only works on 64bit platforms?

File: vm_dump.c [hunk 3]
```diff
@@ -984,6 +991,41 @@ rb_vm_bugreport(const void *ctx)
...
+	mach_msg_type_number_t count = VM_REGION_SUBMAP_INFO_COUNT_64;
+	natural_t depth = 0;
+
+	fprintf(stderr, "* Process memory map:\n\n");
+	while (1) {
+		if (vm_region_recurse_64(mach_task_self(), &addr, &size, &depth,
```
Comment (nobu) [inline vm_dump.c]: Could you adjust indents as the rest.

File: vm_dump.c (hunk 2 above)
Comment (devnexen) [reply vm_dump.c]: Indeed.

File: vm_dump.c [hunk 4]
```diff
@@ -994,36 +997,36 @@ rb_vm_bugreport(const void *ctx)
...
-
-		if (map.is_submap) {
-			// We only look at main addresses
-			depth++;
-		} else {
-			fprintf(stderr, "%lu-%lu %s%s%s", addr, (addr+size),
-				((map.protection & VM_PROT_READ) != 0 ? "r" : "-"),
-				((map.protection & VM_PROT_WRITE) != 0 ? "w" : "-"),
-				((map.protection & VM_PROT_EXECUTE) != 0 ? "x" : "-"));
+        if (vm_region_recurse(mach_task_self(), &addr, &size, &depth,
```
Comment (nobu) [inline vm_dump.c]: Why removed the indents?
You may want to indent by spaces?

Comment (devnexen) [reply vm_dump.c]: you are right sorry.

**Pull request merged**